benchmarks/package.lisp

```diff
@@ -1,40 +1,100 @@
 ;;;; package.lisp
 ;;;;
 ;;;; Benchmarks packages and common functions
+;;;;
+
+;;; Each bechmark group are in separate packages, benchmark-NAME
+;;; and benchmark-NAME/native.  The latter is used to define
+;;; Coalton functions, and the former calls them with benchmarking macros.
+;;;
+;;; The coalton-benchmarks package defines functions that runs each
+;;; individual benchmark group, or all benchmarks at once.
 
-(benchmark:define-benchmark-package #:coalton-benchmarks
-  (:export #:run-benchmarks
+(cl:defpackage #:coalton-benchmarks
+  (:use #:cl
+        #:trivial-benchmark)
+  (:export #:run-benchmark
+           #:run-benchmarks
            #:run-benchmarks-ci))
 
-(cl:defpackage #:coalton-benchmarks/native
+(cl:in-package #:coalton-benchmarks)
+
+(defparameter *all-benchmarks* '())
+
+(defmacro define-coalton-benchmark (name
+                                    benchmark-clauses
+                                    &rest native-clauses)
+  (let ((benchmark-package (intern (format nil "BENCHMARK-~S" name) 'keyword))
+        (native-package (intern (format nil "BENCHMARK-~S/NATIVE" name) 'keyword)))
```

Comment:
Can these be `COALTON-BENCHMARK/...`? It's a little wordier, but worth it.

Maybe also add a default package local nickname "NATIVE" for the native package. That way the actual benchmarks can always refer to `native:...`.

In general I don't like symbol generating macros, but for this limited use case I think it's fine.